mesonbuild/backend/ninjabackend.py

```diff
@@ -2387,6 +2387,21 @@ def generate_dynamic_link_rules(self) -> None:
 
                 options = self._rsp_options(compiler)
                 self.add_rule(NinjaRule(rule, command, args, description, **options, extra=pool))
+                if mesonlib.is_os2():
+                    rule = '{}_DLL_LINKER{}'.format(langname, self.get_rule_suffix(for_machine))
+                    command = compiler.get_linker_exelist()
+                    args =  ['$ARGS'] + NinjaCommandArg.list(compiler.get_linker_output_args('$out'), Quoting.none) + ['$in', '$LINK_ARGS']
+                    args += ['&&', 'emximp', '-o', '$IMPLIB', '$out']
+
+                    description = 'Linking target $out'
+                    if num_pools > 0:
+                        pool = 'pool = link_pool'
+                    else:
+                        pool = None
+
+                    options = self._rsp_options(compiler)
+                    self.add_rule(NinjaRule(rule, command, args, description, **options, extra=pool))
+
```

Comment:
It seems like you always end up using this rule for dynamic linking on OS/2 paths, am I missing something?

Reply:
Yes, in order to build an import library.

Reply:
So can we just have the normal shared_library linking path do this on OS/2, or is there a use of that I missed?

Reply:
I want to do so, but I'm sorry I have no idea how to do. Any hints ?